### ISSUE TITLE
bdev_raid_delete RPC does not clear on-disk metadata

### ISSUE THREAD

**Issue by jimharris** [spdk/spdk#3412]
The bdev_raid_delete RPC should clear on-disk metadata if it exists (i.e. superblock option was defined). Without it, the user is required to find someway to clear that metadata itself.

**Comment by apaszkie** [comment]
I agree that `bdev_raid_delete` should clear the metadata but I think it would be good to have `bdev_raid_start`/`bdev_raid_stop` RPCs first, see the discussion here: https://github.com/spdk/spdk/issues/3341#issuecomment-2072417505. Without that, there would be no way to stop the raid_bdev and keep the metadata. Or maybe we could add an option to `bdev_raid_delete` that would determine if the metadata should be cleared or not?

**Comment by jimharris** [comment]
> I agree that `bdev_raid_delete` should clear the metadata but I think it would be good to have `bdev_raid_start`/`bdev_raid_stop` RPCs first, see the discussion here: [#3341 (comment)](https://github.com/spdk/spdk/issues/3341#issuecomment-2072417505). Without that, there would be no way to stop the raid_bdev and keep the metadata. Or maybe we could add an option to `bdev_raid_delete` that would determine if the metadata should be cleared or not?

Adding an option to `bdev_raid_delete` to clear the metadata sounds OK to me.

**Comment by clay-mayers** [comment]
> > I agree that `bdev_raid_delete` should clear the metadata but I think it would be good to have `bdev_raid_start`/`bdev_raid_stop` RPCs first, see the discussion here: [#3341 (comment)](https://github.com/spdk/spdk/issues/3341#issuecomment-2072417505). Without that, there would be no way to stop the raid_bdev and keep the metadata. Or maybe we could add an option to `bdev_raid_delete` that would determine if the metadata should be cleared or not?
> 
> Adding an option to `bdev_raid_delete` to clear the metadata sounds OK to me.

It doesn't cover all the cases to require a working raid device to delete the base bdev super blocks.  The code that builds the raid devices need to ignore them.  Issue #3562 proposes a flag on create to ignore - I think create should just ignore them always.  It's `raid_bdev_add_base_bdev` that needs an option to ignore the SB.
